Cupcake.cpp

```diff
@@ -0,0 +1,142 @@
+#include "Cupcake.h"
+#include "Player.h"
+
+using namespace sf;
+//Constructor - Destructor
+Cupcake::Cupcake() 
+{
+	srand((int)time(0));
+
+	//Randomize cupcake speed
+	m_cupcakeSpeed = (rand() % 2) + 8;
+	m_maxCupcakes = 5;
+
+	//Initialize bool
+	m_deleted = false;
+	
+}
+
+Cupcake::~Cupcake()
+{
+
+}
+
+
+void Cupcake::update(sf::RenderWindow& w)
+{
+		
+		//Update and reset the timer for spawning cupcakes and moving them
+		m_cupcakeSpawnTimer += .3f;
+
+		//Update the cupcakes as per last frame
+		removeCupcake();
+		
+
+		//spawn and move cupcakes
+		if (m_cupcakeSpawnTimer > m_cupcakeSpawnTimerMax)
+		{
+			spawn();
+			move();
+			m_cupcakeSpawnTimer = 1.f;
+
+		}
+		else
+		{//Give a cooldown between timer resets
+			m_cupcakeSpawnTimer += 0.1f;
+		}
+		
+		
+		
+}
+
+
+
+void Cupcake::spawn()
+{//How many seconds in game time have passed?
+	FrameTimer ft;
+	const double dt = ft.Mark();
+		//Set cupcake sprites origin point to be at the center 
+		m_cupcakeSprite.setOrigin(sf::Vector2f(25.f, 25.f));

```

Comment:
I had not thought of it this way. Yes, it can be a variable contained in this function. Is this better? Why?

Reply:
Exactly! In programming it is generally a good practice to reduce the number of lines between creating something and using something. Since `m_cupcakeSprite` is not really being used, it does not need to exist as a member!